lib/openscaptest.pm

```diff
@@ -0,0 +1,67 @@
+# Copyright (C) 2017-2018 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# Summary: Base module for openSCAP test cases
+# Maintainer: Wes <whdu@suse.com>
+
+package openscaptest;
+
+use strict;
+use testapi;
+use utils;
+use version_utils qw(is_sle is_leap);
+use base 'consoletest';
+
+sub oscap_get_test_file {
+    my ($self, $source_file) = @_;
+
+    assert_script_run 'wget --quiet ' . data_url("openscap/$source_file")
+      if (!get_test_data("$source_file"));
+}
+
+sub ensure_oval_result {
+    my ($self, $oval_file, $result_file) = @_;
+
+    assert_script_run "oscap oval eval --results $result_file $oval_file"
+      if (!get_test_data($result_file));
+}
+
+sub ensure_xccdf_result {
+    my ($self, $oval_file, $result_file) = @_;
+
+    assert_script_run "oscap xccdf eval --profile standard --results $result_file $oval_file"
+      if (!get_test_data($result_file));
+}
+
+sub validate_result {
+    my ($self, $result_file, $match) = @_;
+
+    assert_script_run "xmllint $result_file";
+    validate_script_output "cat $result_file", sub { $match };
+    upload_logs($result_file);
+}
+
+sub pre_run_hook {
+    my ($self) = @_;
+
+    select_console 'root-console';
+
+    zypper_call('in openscap-utils');
+
+    $self->oscap_get_test_file("oval.xml");
+    $self->oscap_get_test_file("xccdf.xml");
```

Comment:
I guess you are missing to submit `openscap/oval.xml` and `openscap/xccdf.xml` in this PR?
Also I think this two lines would be called in every test openscap module, that seems duplicate action. Better to add conditional statements to donwload them once, or you can clean up downloaded xml in every openscap module.

Reply:
Thank you for pointing that, I just added ```data/openscap/oval.xml``` and ```data/openscap/xccdf.xml```
For __$self->oscap_get_test_file__, it has already involved a if condition with __get_test_data__. When it find these file existed, it will not re-download them.

Reply:
`get_test_data` check data file and returns content from `/data` directory and doesn't check if it exists in SUT. From your prove runs download executed in every openscap module.

Reply:
Okay, I have changed the code of oscap_get_test_file to test if file existed before downloading, and updated verification test runs.